src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt

```diff
@@ -263,6 +264,57 @@ class SignServiceImpl(
         )
     }
 
+    override fun signQuoteFromHope(
+        completeQuoteId: UUID,
+        request: SignQuoteFromHopeRequest
+    ): Either<ErrorResponseDto, SignedQuoteResponseDto> {
+        val quote = quoteRepository.find(completeQuoteId)
+            ?: throw QuoteNotFoundException("Quote $completeQuoteId not found when trying to sign")
+
+        if (quote.signedProductId != null) {
+            throw RuntimeException("There is a signed product id ${quote.signedProductId} already")
+        }
+
+        if (quote.memberId == null) {
+            Either.Left(
+                ErrorResponseDto(
+                    ErrorCodes.MEMBER_ID_IS_NOT_PROVIDED,
+                    "no member id connected to quote"
```

Comment:
Not sure what the error is used for, but could be nice to have the id in case it gets logged
```suggestion
                    "No member id connected to quote ${quote.id}"
```